src/lib/x509/x509_ext.cpp

```diff
@@ -25,65 +25,98 @@ namespace Botan {
 
 namespace {
 
-std::unique_ptr<Certificate_Extension> extension_from_oid(const OID& oid) {
-   if(oid == Cert_Extension::Subject_Key_ID::static_oid()) {
-      return std::make_unique<Cert_Extension::Subject_Key_ID>();
-   }
+std::optional<uint32_t> is_sub_element_of(const OID& oid, std::initializer_list<uint32_t> prefix) {
+   const auto& c = oid.get_components();
 
-   if(oid == Cert_Extension::Key_Usage::static_oid()) {
-      return std::make_unique<Cert_Extension::Key_Usage>();
+   if(c.size() != prefix.size() + 1) {
+      return {};
    }
 
-   if(oid == Cert_Extension::Subject_Alternative_Name::static_oid()) {
-      return std::make_unique<Cert_Extension::Subject_Alternative_Name>();
+   if(!std::equal(c.begin(), c.end() - 1, prefix.begin(), prefix.end())) {
+      return {};
    }
 
-   if(oid == Cert_Extension::Issuer_Alternative_Name::static_oid()) {
-      return std::make_unique<Cert_Extension::Issuer_Alternative_Name>();
-   }
+   return c[c.size() - 1];
+}
```

Comment:
Perhaps it makes sense to have this as a member in `OID`?

Reply:
Possibly. I considered it, but for now this is the only place it is needed. If we end up doing a similar optimization elsewhere, that to me would be the time the promote this outside of an anon namespace [*]

[*] Though I'm not sure if I would put this as a member on `OID`, or just in an internal header somewhere. This seems like an odd thing for an application to need.